From 73ade06f7598df7fc94989711abfdf4d9255ebb5 Mon Sep 17 00:00:00 2001
From: Mahesh Mahadevan <r9aadq@freescale.com>
Date: Fri, 16 Dec 2011 06:39:38 -0600
Subject: ENGR00170508 Need to zero REG_BYPASS_CNT on exit from STOP mode

Per e-mail from design team, the count needs to be zeroed and
reconfigured on exit from low power mode

Signed-off-by: Mahesh Mahadevan <r9aadq@freescale.com>
---
 arch/arm/mach-mx6/pm.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/mach-mx6/pm.c b/arch/arm/mach-mx6/pm.c
index bdf3371..381cc1d 100644
--- a/arch/arm/mach-mx6/pm.c
+++ b/arch/arm/mach-mx6/pm.c
@@ -161,6 +161,10 @@ static void mx6_suspend_restore(void)
 	/* restore settings after suspend */
 	__raw_writel(anatop[0], anatop_base + ANATOP_REG_2P5_OFFSET);
 	__raw_writel(anatop[1], anatop_base + ANATOP_REG_CORE_OFFSET);
+	/* Per spec, the count needs to be zeroed and reconfigured on exit from
+	 * low power mode
+	 */
+	__raw_writel(ccm_ccr & ~MXC_CCM_CCR_REG_BYPASS_CNT_MASK, MXC_CCM_CCR);
 	udelay(50);
 	__raw_writel(ccm_ccr, MXC_CCM_CCR);
 	__raw_writel(ccm_clpcr, MXC_CCM_CLPCR);
-- 
1.7.9.5

